### DISCUSSION TITLE
Is Slick still being worked on / maintained

### DISCUSSION THREAD

**Discussion by rleinfellner** [slick/slick#2515] [Questions] [answered]
Its seems to have been quite some time since Slick was updated

Is it still being worked on? Will there be a version for Scala 3.X?

If the answer is "sadly not" any ideas of what would be safe alternative for Scala DB interactions?

**Comment by nafg** [comment, marked as answer]
I try to maintain Slick in my free time on Sundays, but help would be very much appreciated.

I'm trying to release 3.4.0 very soon, and then Scala 3 support is my top priority.

For Scala 3 support there's already a branch. First, @szeiger created the first version in https://github.com/szeiger/slick/tree/dottyquery. Then @alexFrankfurt did a lot of work getting it production ready, in https://github.com/slick/slick/pull/2187. I've been keeping it rebased on `main` frequently, last pushed in https://github.com/slick/slick/tree/nafg/dottyquery.

Once 3.4.0 is released I want to rework it so that functionality is not lost for Scala 2, by using scala-version-specific directories where necessary.

**Comment by nafg** [comment]
If you want to help get in touch please.

**Reply by slovvik** [reply]
@nafg I could help.

**Reply by nafg** [reply]
Amazing! Are you on Discord?

**Reply by slovvik** [reply]
slovvik#7234

**Comment by nafg** [comment]
I'm converting this to a Discussion since that's really what it is.

**Comment by sfali** [comment]
Hi There, is there any plan to have Scala 3 support completed. I am not entirely familiar with Slick internal but happy to help.

**Comment by nafg** [comment]
@sfali Yes, it's coming as soon as possible. Keep an eye on https://github.com/slick/slick/tree/nafg/dottyquery for now.

Also see https://github.com/slick/slick/issues/2177

**Comment by nafg** [comment]
In particular, as I said there, I am better at getting things done when I have an interactive audience, so if someone wants to watch me working on it that will help make it happen faster G-d willing.

**Reply by flavius** [reply]
Where can I watch you? I don't have scala experience but it's my main perfect-candidate for the type of applications I do (e-commerce, working as CTO/Architect).

**Reply by nafg** [reply]
Something came up recently and I'm not sure when I'm doing it next, but I usually post on Discord, on the Scala server in the slick channel, as well as the Learning Scala server. I usually either stream on Discord or do it over Google Meet. Does that work for you?

**Reply by golangcyberdude** [reply]
Discord or Meet works for me.  I can help with code reviews and testing.  Still learning Scala, but at A2/L1 (perhaps just barely :) )

Probably should set up on separate VM from my main at work VM.  Scala 3.1.2 below.

```=->git status
On branch nafg/dottyquery
Your branch is up to date with 'origin/nafg/dottyquery'.

[info] Fetched artifacts of
[warn] -- Warning: /home/glcd/slick/slick/src/main/scala-3/slick/lifted/ShapedValue.scala:93:53
[warn] 93 |  implicit def someToTuple[T]: ToTuple[Some[T], T] = identity
[warn]    |                                                     ^^^^^^^^
[warn]    |method identity is eta-expanded even though slick.lifted.ToTuple[Some[T], T] does not have the @FunctionalInterface annotation.
[warn] -- Warning: /home/glcd/slick/slick/src/main/scala-3/slick/lifted/ShapedValue.scala:97:57
[warn] 97 |  implicit def optionToTuple[T]: ToTuple[Option[T], T] = identity
[warn]    |                                                         ^^^^^^^^
[warn]    |method identity is eta-expanded even though slick.lifted.ToTuple[Option[T], T] does not have the @FunctionalInterface annotation.
[warn] -- Warning: /home/glcd/slick/slick/src/main/scala/slick/jdbc/JdbcModelBuilder.scala:278:43
[warn] 278 |        dbType.map(SqlProfile.ColumnOption.SqlType) ++
[warn]     |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[warn]     |The method `apply` is inserted. The auto insertion will be deprecated, please write `slick.sql.SqlProfile.ColumnOption.SqlType.apply` explicitly.
[warn] -- Feature Warning: /home/glcd/slick/slick/src/main/scala/slick/jdbc/StaticQuery.scala:22:15
[warn] 22 |  implicit def typedParameter[T](param: T)(implicit pconv: SetParameter[T]): TypedParameter[T] =
[warn]    |               ^
[warn]    |Definition of implicit conversion method typedParameter should be enabled
[warn]    |by adding the import clause 'import scala.language.implicitConversions'
[warn]    |or by setting the compiler option -language:implicitConversions.
[warn]    |See the Scala docs for value scala.language.implicitConversions for a discussion
[warn]    |why the feature should be explicitly enabled.
[error] -- [E006] Not Found Error: /home/glcd/slick/slick/src/main/scala/slick/memory/MemoryQueryingProfile.scala:97:33
[error] 97 |      def update(value: Any, pr: Updater) = ??
[error]    |                                 ^^^^^^^
[error]    |                                 Not found: type Updater
[error] -- [E006] Not Found Error: /home/glcd/slick/slick/src/main/scala/slick/memory/MemoryQueryingProfile.scala:98:30
[error] 98 |      def set(value: Any, pp: Writer, offset: Int) = ??
[error]    |                              ^^^^^^
[error]    |                              Not found: type Writer
[warn] four warnings found
[error] two errors found
[error] (slick / Compile / compileIncremental) Compilation failed
[error] Total time: 40 s, completed Jan 25, 2023 11:59:37 PM
```

**Comment by nafg** [comment]
Releasing M2 now with dropped scala 2.12, fixed warnings, and `-Xsource:3`

**Comment by nafg** [comment]
3.5.0-M4 is now on maven central, cross-published for 2.12, 2.13, and 3.2.2!
